pkg/apply/ResourceInfos_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"fmt"
+	"sort"
+	"testing"
+
+	"gotest.tools/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+var configMapObj = unstructured.Unstructured{
+	Object: map[string]interface{}{
+		"apiVersion": "v1",
+		"kind":       "ConfigMap",
+		"metadata": map[string]interface{}{
+			"name":      "the-map",
+			"namespace": "testspace",
+		},
+	},
+}
+
+var namespaceObj = unstructured.Unstructured{
+	Object: map[string]interface{}{
+		"apiVersion": "v1",
+		"kind":       "Namespace",
+		"metadata": map[string]interface{}{
+			"name": "testspace",
+		},
+	},
+}
+
+var deploymentObj = unstructured.Unstructured{
+	Object: map[string]interface{}{
+		"apiVersion": "v1",
+		"kind":       "Deployment",
+		"metadata": map[string]interface{}{
+			"name":      "testdeployment",
+			"namespace": "testspace",
+		},
+	},
+}
+
+func TestResourceOrdering(t *testing.T) {
+	configMapInfo := resource.Info{
+		Name:   "the-map",
+		Object: &configMapObj,
+	}
+
+	namespaceInfo := resource.Info{
+		Name:   "testspace",
+		Object: &namespaceObj,
+	}
+
+	deploymentInfo := resource.Info{
+		Name:   "testdeployment",
+		Object: &deploymentObj,
+	}
+
+	infos := []*resource.Info{&deploymentInfo, &configMapInfo, &namespaceInfo}
+	sort.Sort(ResourceInfos(infos))
+
+	var result string
+	for _, info := range infos {
+		result += fmt.Sprintf("Name: %s, Kind: %s\n", info.Name, info.Object.GetObjectKind().GroupVersionKind().Kind)
```

Comment:
Can we find a better way to do the comparison here? Printing it and then comparing strings seems a bit strange.

Reply:
I see that Sprintf doesn't print but just forms and returns the string. Am I missing something.

// Sprintf formats according to a format specifier and returns the resulting string.

Reply:
Yeah, so it doesn't print to Stdout. I meant that turning the results into strings and the comparing it seems strange. Can we just verify that the order of the types follow what is expected?